tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/injected/ManagedThreadFactoryDefinitionOnEJBInjectedServlet.java

```diff
@@ -0,0 +1,247 @@
+            assertEquals(thread1.getPriority(), 4, "New threads must be created with the priority that is specified on "
+                    + "ManagedThreadFactoryDefinition");
+
+            BlockingQueue<Object> results = new LinkedBlockingQueue<>();
+
+            threadFactoryA.newThread(() -> {
+                results.add(Thread.currentThread().getPriority());
+                results.add(StringContext.get());
+                results.add(IntContext.get());
+                try {
+                    results.add(InitialContext.doLookup("java:app/concurrent/ContextA"));
+                } catch (Throwable x) {
+                    results.add(x);
+                }
+            }).start();
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(4),
+                    "ManagedThreadFactory must start threads with the configured priority.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), "",
+                    "Third-party context type StringContext must be cleared from thread "
+                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(161),
+                    "Third-party context type IntContext must be propagated to thread "
+                            + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration "
+                            + "based on the thread context at the time the ManagedThreadFactory was looked up.");
+
+            Object lookupResult = results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+            if (lookupResult instanceof Throwable)
+                throw new AssertionError().initCause((Throwable) lookupResult);
+            assertTrue(lookupResult instanceof ContextService, "Application context must be propagated to thread "
+                    + "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration.");
+        } finally {
+            IntContext.set(0);
+            StringContext.set("");
+        }
+    }
+
+    /**
+     * ManagedThreadFactoryDefinition defined on an EJB with minimal attributes
+     * creates threads with normal priority and uses java:comp/DefaultContextService
+     * to determine context propagation and clearing.
+     */
+    public void testManagedThreadFactoryDefinitionDefaultsEJB() throws Throwable {
+        CountDownLatch blocker = new CountDownLatch(1);
+        CountDownLatch allThreadsRunning = new CountDownLatch(2);
+        CompletableFuture<Object> lookupTaskResult = new CompletableFuture<>();
+        CompletableFuture<Object> txTaskResult = new CompletableFuture<>();
+
+        Runnable lookupTask = () -> {
+            try {
+                allThreadsRunning.countDown();
+                blocker.await(MAX_WAIT_SECONDS * 5, TimeUnit.SECONDS);
+                lookupTaskResult.complete(threadFactoryB);
```

Comment:
The Runnable named "lookupTask" wasn't actually performing a lookup, just returning an existing value, which didn't test context propagation at all.
```suggestion
                lookupTaskResult.complete(InitialContext.doLookup("java:comp/concurrent/EJBThreadFactoryB"));
```